cosmoz-omnitable-column-mixin.html

```diff
@@ -0,0 +1,545 @@
+				}
+
+				get headerTemplatizer() {
+					return this._headerTemplatizer;
+				}
+
+				getHeaderTemplateInstance() {
+					if (
+						!this._headerTemplatizer ||
+						!this._headerTemplatizer.createInstance
+					) {
+						return;
+					}
+					return this._headerTemplatizer.createInstance();
+				}
+
+				releaseHeaderTemplateInstance(instance) {
+					if (this._headerTemplatizer) {
+						this._headerTemplatizer.releaseInstance(instance);
+					}
+				}
+
+				/**
+				 * Override this in column elements if you need a different default width
+				 */
+
+				_computePreferredDropdownHorizontalAlign(columnIndex) {
+					return columnIndex === 0 ? 'left' : 'right';
+				}
+				getString(item, valuePath = this.valuePath) {
+					if (valuePath === undefined) {
+						console.error(
+							this,
+							'has undefined valuePath arg',
+							valuePath,
+							'for item',
+							item
+						);
+						return;
+					}
+					return this.get(valuePath, item);
+				}
+				toXlsxValue(item, valuePath = this.valuePath) {
+					if (!valuePath) {
+						return '';
+					}
+					return this.get(valuePath, item);
+				}
+
+				_pathsChanged(valuePath, groupOn, sortOn) {
+					if (valuePath == null) {
+						return;
+					}
+
+					if (groupOn == null) {
+						this.set('groupOn', valuePath);
+					}
+
+					if (sortOn == null) {
+						this.set('sortOn', valuePath);
+					}
+				}
+
+				getComparableValue(item, valuePath) {
+					return this.get(valuePath, item);
+				}
+
+				_valueChanged(event) {
+					var input = event.target,
+						value = input.value,
+						item = event.model.item,
+						oldValue = this.get(this.valuePath, item);
+
+					this.set(this.valuePath, value, item);
+					this._fireItemChangeEvent(item, this.valuePath, oldValue);
+				}
+
+				_fireItemChangeEvent(item, valuePath, oldValue, formatFn) {
+					var value = this.get(valuePath, item),
+						change = {
+							item,
+							valuePath: this.valuePath,
+							value: formatFn ? formatFn(value) : value,
+							oldValue: formatFn ? formatFn(oldValue) : oldValue
+						};
+
+					this.dispatchEvent(new CustomEvent('column-item-changed', {detail: change}));
+				}
+
+				getFilterFn() {
+					if (!this.filter) {
+						return;
+					}
+					if (!Array.isArray(this.filter)) {
+						return this._applySingleFilter.bind(
+							this,
+							this.filter.toString().toLowerCase()
+						);
+					}
+					if (this.filter.length === 0) {
+						return;
+					}
+					return this._applyMultiFilter.bind(this, this.filter);
+				}
+
+				resetFilter() {
+					this.filter = this._getDefaultFilter();
+				}
+
+				/**
+				 * Returns whether `filterNotify.base` or `filter` is set to a usable value.
+				 * @param {Object|void} filterNotify filter.*
+				 * @returns {Boolean}  True if filter or filterNotify.base is set
+				 */
+				hasFilter(filterNotify) {
+					const filter = filterNotify == null ? this.filter : filterNotify.base,
+						isArray = Array.isArray(filter);
+
+					if (
```

Comment:
When if statements go on multiple lines, that's an indication they're unnecessarily complex.
isArray will return false if filter is null or '' so no need for those checks.
You can split it up into one if for arrays and one for objects to make it more readable.
```suggestion
					if (Array.isArray(filter)) {
						return filter.length > 0;
					}
					if (typeof filter === 'object')  {
						const setFilters = Object.keys(filter).filter(k => filter[k] != null);
						return setFilters.length > 0;
					}
```